### PR TITLE
symbolic function implementation for solveset

### PR DESCRIPTION
**Pull request by Yathartha22**
This is in reference to #12239 
I tried to solve `symbolic functions` by replacing `functions` to `dummy` symbols and then solving them as normal expressions.

### CODE REVIEW

**Comment by Yathartha22** [comment]
```
$ python ./bin/test solvers
Traceback (most recent call last):
  File "./bin/test", line 105, in <module>
    import sympy
  File "C:\Users\DELL\Desktop\sympy\sympy\__init__.py", line 64, in <module>
    from .concrete import *
  File "C:\Users\DELL\Desktop\sympy\sympy\concrete\__init__.py", line 2, in <module>
    from .summations import summation, Sum
  File "C:\Users\DELL\Desktop\sympy\sympy\concrete\summations.py", line 10, in <module>
    from sympy.calculus.singularities import is_decreasing
  File "C:\Users\DELL\Desktop\sympy\sympy\calculus\__init__.py", line 6, in <module>
    from .singularities import (singularities, is_increasing,
  File "C:\Users\DELL\Desktop\sympy\sympy\calculus\singularities.py", line 19, in <module>
    from sympy.solvers.solveset import solveset
  File "C:\Users\DELL\Desktop\sympy\sympy\solvers\__init__.py", line 15, in <module>
    from .diophantine import diophantine
  File "C:\Users\DELL\Desktop\sympy\sympy\solvers\diophantine.py", line 27, in <module>
    from sympy.solvers.solveset import solveset_real
  File "C:\Users\DELL\Desktop\sympy\sympy\solvers\solveset.py", line 851
    else
       ^
SyntaxError: invalid syntax

```
when I am trying to test the module I am receiving such message. They were testing okay, but suddenly it started to give such message.
This is the reason why Travis builds are failing. I am not able to figure out the problem.

**Comment by roger-** [comment]
As a longer term fix, shouldn't symbolic functions be derived from normal symbols  so they have the same behaviors (vs having the explicit check)?

**Comment by smichr** [comment]
The last line of the printout is telling you that there is a syntax error; you are missing the colon after `else` as in `else:`.

**Comment by Yathartha22** [comment]
@smichr thanks for the bug fix. I don't know how I was not able to see the bug. I will fix this

**Comment by Yathartha22** [comment]
@smichr also, could you give your suggestions about the idea behind this implementation. Is this a good approach or is there anything that I skipped while implementing

**Comment by Yathartha22** [comment]
@roger could you ellaborate

**Comment by roger-** [comment]
I don't know the SymPy internals but symbolic functions seem like they should inherit from regular symbols making any special check unnecessary.

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -843,6 +843,18 @@ def solveset(f, symbol=None, domain=S.Complexes):
     if not isinstance(f, (Expr, Number)):
         raise ValueError("%s is not a valid SymPy expression" % (f))
 
+    if isinstance(symbol, Function):
+        f_tup = f.args
+        for arg in f_tup:
+            if arg.has(symbol) or arg.is_Number:
+                continue
+            else:
+                raise ValueError("%s is not a valid expression" % (f))
```
Comment (ylemkimon) [inline sympy/solvers/solveset.py]: It doesn't mean that it is not a valid expression, rather it means that it is not solvable using this method and should be passed to other solvers.

**Pull request closed without merging**